x-pack/legacy/plugins/rollup/__jest__/client_integration/job_list_clone.test.js

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { setupEnvironment, pageHelpers, nextTick } from './helpers';
+import { JOB_TO_CLONE, JOB_CLONE_INDEX_PATTERN_CHECK } from './helpers/constants';
+
+jest.mock('ui/index_patterns', () => {
+  const { INDEX_PATTERN_ILLEGAL_CHARACTERS_VISIBLE } = require.requireActual(
+    '../../../../../../src/legacy/ui/public/index_patterns/constants'
+  ); // eslint-disable-line max-len
+  return { INDEX_PATTERN_ILLEGAL_CHARACTERS_VISIBLE };
+});
+
+jest.mock('ui/chrome', () => ({
+  addBasePath: () => '/api/rollup',
+  breadcrumbs: { set: () => {} },
+  getInjected: () => ({}),
+}));
+
+jest.mock('lodash/function/debounce', () => fn => fn);
+
+const { setup } = pageHelpers.jobList;
+
+describe('Cloning an existing rollup job from job list', () => {
+  let server;
+  let httpRequestsMockHelpers;
+  let table;
+  let find;
+  let component;
+  let exists;
+
+  beforeAll(() => {
+    ({ server, httpRequestsMockHelpers } = setupEnvironment());
+  });
+
+  afterAll(() => {
+    server.restore();
+  });
+
+  beforeEach(async () => {
+    httpRequestsMockHelpers.setIndexPatternValidityResponse(JOB_CLONE_INDEX_PATTERN_CHECK);
+    httpRequestsMockHelpers.setLoadJobsResponse(JOB_TO_CLONE);
+
+    ({ find, exists, table, component } = setup());
+
+    await nextTick(); // We need to wait next tick for the mock server response to comes in
+    component.update();
+  });
+
+  it('should navigate to create view with default values set', async () => {
+    const { rows } = table.getMetaData('rollupJobsListTable');
+    const button = rows[0].columns[1].reactWrapper.find('button');
+
+    expect(exists('rollupJobDetailFlyout')).toBe(false); // make sure it is not shown
+
+    button.simulate('click');
+
+    expect(exists('rollupJobDetailFlyout')).toBe(true);
+    expect(exists('jobActionMenuButton')).toBe(true);
+
+    find('jobActionMenuButton').simulate('click');
+    find('jobActionMenuButton').simulate('click');
+    find('jobCloneActionContextMenu').simulate('click');
```

Comment:
So I may be missing something, but I couldn't find elements of the create form (looked for title) and I guess that's because router.getHistory has been mocked out so the create form isn't actually rendering,

Reply:
Ah ok. I'm not as familiar with the rollup tests. @sebelga might be able to help here.

It just seemed a bit strange to me that we were simulating a click, but then there was no `expect` statement. Maybe something like this would be sufficient?

```
expect(exists('jobCloneActionContextMenu')).toBe(true);
```

Reply:
I don't remember mocking the `router.getHistory()`. We should be able to navigate to internal pages. Have you console.log the `component.debug()` html and try to see what was happening in the html?

Reply:
@sebelga Nope, thanks will give that a try!

Reply:
@jloleysens Sorry I overlooked it. If you need to test the change of page, then we need to test 1 level higher (the component that holds the `<Route />` definitions).
It will get tricky so I am not sure the coverage it would add compensate the effort.

Reply:
@sebelga :+1: I've updated the expectations for the test to be a bit more sensible. Not expanding the tests to cover actual navigation. Thanks for clarifying!